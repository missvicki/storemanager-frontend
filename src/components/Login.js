import React from 'react';
import { Message, Header, Form, Grid, Segment } from 'semantic-ui-react';

import '../styles/login.css'

const Login = (props) => {
    const {
        user_name,
        password,
        loginError,
        onSubmitHandler,
        onChangeHandler,
        onDismissHandler,
        loading
    } = props
    const loginErorMsg = (
        <Message
            negative
            content={loginError}
            onDismiss={onDismissHandler}
            style={{ width: "50%", margin: "auto", color: "red" }}
        />
    );

    return (
        <div style={{ padding: "10% 0" }} className="loginPage">
            <Grid columns={2} stackable>
                <Grid.Column width={4}></Grid.Column>
                <Grid.Column width={8} padded="very">
                    <Segment padded="very" color="teal">
                        <Header as="h1" textAlign="center" padded="very">Store Manager</Header>
                        <Header as="h6" style={{ color: "#008080", marginTop: "5px", textAlign: "center", fontSize: "1em" }}>
                            Log in to your account
                        </Header>
                        <br />
                        {loginError ? loginErorMsg : null}
                        <br />
                        <center>
                            <Form onSubmit={onSubmitHandler}>
                                <Form.Field>
                                    <input
                                        type="text"
                                        placeholder="Username"
                                        name="user_name"
                                        value={user_name}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Field>
                                    <input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={password}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Button type="submit" style={{ background: "teal", color: "white" }} loading={loading}>Sign in</Form.Button>
                            </Form>
                        </center>
                    </Segment>

                </Grid.Column>
            </Grid>

        </div>
    )
};

export default Login;
