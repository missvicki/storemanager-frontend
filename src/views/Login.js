import React, { Component } from 'react';
import { connect } from "react-redux";

import Login from '../components/Login';
import loginFetch from "../actions/loginAction"
import { removeLoginError } from "../actions/index";

export class LoginView extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user_name: "",
            password: "",
            role: "admin"
        };
    }

    onSubmitHandler = e => {
        e.preventDefault();
        const { user_name, password, role } = this.state;
        const data = { user_name, password, role };
        const url =
            "https://store-manager-ap1.herokuapp.com/api/v2/auth/login";
        const payload = {
            data,
            history: this.props.history,
            url
        };
        this.props.login(payload);
    };

    onDismissHandler = () => {
        const {
            dismissLoginError
        } = this.props
        dismissLoginError();
    };

    onChangeHandler = e => this.setState({ [e.target.name]: e.target.value });

    render() {
        const loginProps = {
            user_name: this.state.user_name,
            password: this.state.password,
            role: this.state.role,
            loginError: this.props.loginError,
            onSubmitHandler: this.onSubmitHandler,
            onChangeHandler: this.onChangeHandler,
            onDismissHandler: this.onDismissHandler,
            loading: this.props.loading
        };
        return (
            <Login {...loginProps} />
        );
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        login: payload => dispatch(loginFetch(payload)),
        dismissLoginError: () => dispatch(removeLoginError())
    };
};

export const mapStateToProps = state => {
    return {
        loginError: state.login.loginError,
        loading: state.login.loading
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginView);
