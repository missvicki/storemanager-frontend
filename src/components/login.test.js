import React from 'react';
import { mount } from 'enzyme';

import Login from './Login';

describe('Login', () => {
    const onChangeHandler = jest.fn();
    const onDismissHandler = jest.fn();
    const onSubmitHandler = jest.fn();

    const loginProps = {
        user_name: "",
        password: "",
        onDismissHandler: onDismissHandler,
        loginError: "",
        loading:false,
        onSubmitHandler: onSubmitHandler,
        onChangeHandler: onChangeHandler
    }
    const login = mount(<Login {...loginProps}/>)
    
    it('should render the Login Component', () => {
        expect(login).toMatchSnapshot();
    });

    it("calls onSubmitHandler when form is submited", () => {
        login.find("form").simulate("submit");
        expect(onSubmitHandler).toHaveBeenCalled();
    });

    it("calls onChandeHandler when the username or password input is changed", () => {
        login.find('input[name="user_name"]').simulate("change");
        expect(onChangeHandler).toHaveBeenCalled();
        login.find('input[name="password"]').simulate("change");
        expect(onChangeHandler).toHaveBeenCalled();
    });
});
