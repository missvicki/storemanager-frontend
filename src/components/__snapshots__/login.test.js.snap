// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login Component 1`] = `
<Login
  loading={false}
  loginError=""
  onChangeHandler={[MockFunction]}
  onDismissHandler={[MockFunction]}
  onSubmitHandler={[MockFunction]}
  password=""
  user_name=""
>
  <div
    className="loginPage"
    style={
      Object {
        "padding": "10% 0",
      }
    }
  >
    <Grid
      columns={2}
      stackable={true}
    >
      <div
        className="ui stackable two column grid"
      >
        <GridColumn
          width={4}
        >
          <div
            className="four wide column"
          />
        </GridColumn>
        <GridColumn
          padded="very"
          width={8}
        >
          <div
            className="eight wide column"
            padded="very"
          >
            <Segment
              color="teal"
              padded="very"
            >
              <div
                className="ui teal very padded segment"
              >
                <Header
                  as="h1"
                  padded="very"
                  textAlign="center"
                >
                  <h1
                    className="ui center aligned header"
                    padded="very"
                  >
                    Store Manager
                  </h1>
                </Header>
                <Header
                  as="h6"
                  style={
                    Object {
                      "color": "#008080",
                      "fontSize": "1em",
                      "marginTop": "5px",
                      "textAlign": "center",
                    }
                  }
                >
                  <h6
                    className="ui header"
                    style={
                      Object {
                        "color": "#008080",
                        "fontSize": "1em",
                        "marginTop": "5px",
                        "textAlign": "center",
                      }
                    }
                  >
                    Log in to your account
                  </h6>
                </Header>
                <br />
                <br />
                <center>
                  <Form
                    as="form"
                    onSubmit={[MockFunction]}
                  >
                    <form
                      className="ui form"
                      onSubmit={[Function]}
                    >
                      <FormField>
                        <div
                          className="field"
                        >
                          <input
                            name="user_name"
                            onChange={[MockFunction]}
                            placeholder="Username"
                            type="text"
                            value=""
                          />
                        </div>
                      </FormField>
                      <FormField>
                        <div
                          className="field"
                        >
                          <input
                            name="password"
                            onChange={[MockFunction]}
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                        </div>
                      </FormField>
                      <FormButton
                        as={[Function]}
                        control={[Function]}
                        loading={false}
                        style={
                          Object {
                            "background": "teal",
                            "color": "white",
                          }
                        }
                        type="submit"
                      >
                        <FormField
                          control={[Function]}
                          loading={false}
                          style={
                            Object {
                              "background": "teal",
                              "color": "white",
                            }
                          }
                          type="submit"
                        >
                          <div
                            className="field"
                          >
                            <Button
                              as="button"
                              loading={false}
                              style={
                                Object {
                                  "background": "teal",
                                  "color": "white",
                                }
                              }
                              type="submit"
                            >
                              <Ref
                                innerRef={
                                  Object {
                                    "current": <button
                                      class="ui button"
                                      style="background: teal; color: white;"
                                      type="submit"
                                    >
                                      Sign in
                                    </button>,
                                  }
                                }
                              >
                                <RefFindNode
                                  innerRef={
                                    Object {
                                      "current": <button
                                        class="ui button"
                                        style="background: teal; color: white;"
                                        type="submit"
                                      >
                                        Sign in
                                      </button>,
                                    }
                                  }
                                >
                                  <button
                                    className="ui button"
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "background": "teal",
                                        "color": "white",
                                      }
                                    }
                                    type="submit"
                                  >
                                    Sign in
                                  </button>
                                </RefFindNode>
                              </Ref>
                            </Button>
                          </div>
                        </FormField>
                      </FormButton>
                    </form>
                  </Form>
                </center>
              </div>
            </Segment>
          </div>
        </GridColumn>
      </div>
    </Grid>
  </div>
</Login>
`;
